ui/pages/settings/networks-tab/networks-form/rpc-url-editor.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef, useState } from 'react';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  Icon,
+  IconName,
+  IconSize,
+  Popover,
+  PopoverPosition,
+  Text,
+} from '../../../../components/component-library';
+import {
+  AlignItems,
+  BorderColor,
+  BorderRadius,
+  Display,
+  IconColor,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export const RpcUrlEditor = () => {
+  const t = useI18nContext();
+  const [isOpen, setIsOpen] = useState(false);
+  const rpcDropdown = useRef(null);
+
+  // TODO: real endpoints
+  const dummyRpcEndpoints = [
+    'https://palmn-mainnet.infura.io',
+    'https://palm-mainnet.public.blastapi.io',
+    'https://tatum.io/v3/blockchain/node/palm',
+  ];
+
+  return (
+    <>
+      <Text
+        className="networks-tab__rpc-header"
+        marginTop={1}
+        marginBottom={1}
+        variant={TextVariant.bodySmBold}
+      >
+        {t('defaultRpcUrl')}
+      </Text>
+      <Box
+        onClick={() => setIsOpen(!isOpen)}
+        className="networks-tab__rpc-text"
+        display={Display.Flex}
+        justifyContent={JustifyContent.spaceBetween}
+        borderRadius={BorderRadius.MD}
+        borderColor={BorderColor.borderDefault}
+        borderWidth={1}
+        padding={2}
+        ref={rpcDropdown}
+      >
+        <Text variant={TextVariant.bodySm}>{dummyRpcEndpoints[0]}</Text>
+        <Icon name={isOpen ? IconName.ArrowUp : IconName.ArrowDown} />
+      </Box>
+      <Popover
```

Comment:
The width of this popover is just based on contents for now.  Will need to be constrained once this form becomes a fullscreen/modal.